webui/react/src/components/ContinueTrial.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { forwardRef, useCallback, useImperativeHandle, useState } from 'react';
+
+import CreateExperimentModal, { CreateExperimentType } from 'components/CreateExperimentModal';
+import handleError, { ErrorType } from 'ErrorHandler';
+import { paths, routeToReactUrl } from 'routes/utils';
+import { createExperiment } from 'services/api';
+import { ExperimentBase, RawJson, TrialDetails, TrialHyperParameters } from 'types';
+import { clone } from 'utils/data';
+import { trialHParamsToExperimentHParams, upgradeConfig } from 'utils/types';
+
+export interface ContinueTrialHandles {
+  show: () => void;
+}
+
+interface Props {
+  experiment: ExperimentBase;
+  ref?: React.Ref<ContinueTrialHandles>;
+  trial: TrialDetails;
+}
+
+const trialContinueConfig = (
+  experimentConfig: RawJson,
+  trialHparams: TrialHyperParameters,
+  trialId: number,
+): RawJson => {
+  return {
+    ...experimentConfig,
+    hyperparameters: trialHParamsToExperimentHParams(trialHparams),
+    searcher: {
+      max_length: experimentConfig.searcher.max_length,
+      metric: experimentConfig.searcher.metric,
+      name: 'single',
+      smaller_is_better: experimentConfig.searcher.smaller_is_better,
+      source_trial_id: trialId,
+    },
+  };
+};
+
+const ContinueTrial: React.FC<Props> = forwardRef(function ContinueTrial(
+  { experiment, trial }: Props,
+  ref?: React.Ref<ContinueTrialHandles>,
+) {
+  const [ contModalConfig, setContModalConfig ] = useState<RawJson>();
+  const [ contModalError, setContModalError ] = useState<string>();
+  const [ isVisible, setIsVisible ] = useState(false);
+
+  const show = useCallback(() => {
+    const rawConfig = trialContinueConfig(clone(experiment.configRaw), trial.hparams, trial.id);
+    let newDescription = `Continuation of trial ${trial.id}, experiment ${trial.experimentId}`;
+    if (rawConfig.description) newDescription += ` (${rawConfig.description})`;
+    rawConfig.description = newDescription;
+    upgradeConfig(rawConfig);
+    setContModalConfig(rawConfig);
+    setIsVisible(true);
+  }, [ experiment.configRaw, trial ]);
```

Comment:
**question:** there might be special circumstances for doing it this way, but thought I ask this:
What if you pass through the `visible` on `CreateExperimentModal` to be a prop on `ContinueTrial` and use an `useEffect` + `usePrevious` to detect when `visible goes from `false` to `true` to do the `rawConfig` processing that's done in this `show` block.

Reply:
I think this adds un-needed complexity due to the fact that we need to pass both the state and the setState to the component and handle detect the status change.

But if you prefer that it's easy to change.
Pls lmk.

Reply:
as is fine, thanks!